ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Update.hs

```diff
@@ -33,17 +36,72 @@ import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog
 import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog.Update
 import           Ouroboros.Consensus.Storage.LedgerDB.ReadsKeySets
 import           Ouroboros.Consensus.Util.IOLike
+import qualified Ouroboros.Network.AnchoredSeq as AS
+
+data LedgerDBUpdate = Prune | Extend
+  deriving (Show)
 
 -- | PRECONDITION: The new 'LedgerDB' must be the result of calling either
 -- 'LedgerDB.ledgerDbSwitch' or 'LedgerDB.ledgerDbPushMany' on the current
 -- 'LedgerDB'.
 setCurrent ::
-     MonadSTM m
+     forall blk m.
+     ( MonadSTM m
+     , IsLedger (LedgerState blk)
+     , HasLedgerTables (LedgerState blk)
+     , HasCallStack
+     )
   => StrictTVar m (DbChangelog' blk)
   -> DbChangelog' blk
+  -> LedgerDBUpdate
   -> STM m ()
-setCurrent = writeTVar
+setCurrent v dblog = \case
+  Prune ->
+    updatePruning (changelogLastFlushedState dblog)
+  Extend ->
+    updateExtending (changelogVolatileStates dblog) (changelogDiffs dblog)
+ where
+   updatePruning :: ExtLedgerState blk EmptyMK -> STM m ()
```

Comment:
Can two `updatePruning` operations race with each other? Are there any implicit assumptions here? For instance, I wonder if we are assuming that we prune only in one place (in the copy and snapshot thread).